lib/compress/zstd_compress.c

```diff
@@ -2914,6 +3006,14 @@ static size_t ZSTD_buildSeqStore(ZSTD_CCtx* zc, const void* src, size_t srcSize)
         } else if (zc->appliedParams.ldmParams.enableLdm == ZSTD_ps_enable) {
             rawSeqStore_t ldmSeqStore = kNullRawSeqStore;
 
```

Comment:
I'm still lost here.
LDM seems more clearly enabled in this block.
So what was happening in the previous one ?

Reply:
No idea :) I disabled the external matchfinder in both of the `ZSTD_ldm_blockCompress` branches.

Reply:
I discussed this with @terrelln. The previous branch is for multi-thread LDM specifically. It's in a separate branch because LDM is a serial step in multi-thread compression. The process for a child CCtx to integrate the sequences from that serial step is a little different than the process where a single CCtx does both the LDM and normal matchfinding.

The previous branch has `assert(zc->appliedParams.ldmParams.enableLdm == ZSTD_ps_disable);` because we want to make sure the *child* CCtx isn't using the single-thread LDM code path. However, the parent ZSTDMT_CCtx will have LDM enabled when that branch runs.

In terms of this PR, what this all means is that we can safely fail compression in these branches when the external matchfinder is enabled, and it will *only* affect LDM compression. Other params won't be impacted, e.g. block splitter.